src/app/shared/services/transaction.service.ts

```diff
@@ -38,23 +43,31 @@ export class TransactionService implements OnDestroy {
 
   addTransaction(transaction: Transaction): void {
     this.currentTransactions = [ ...this.currentTransactions, transaction ];
-    this.saveToStore(this.currentTransactions);
+    this.trackedTransactionHashes.push(transaction.transactionHash);
```

Comment:
I read somewhere that from angular's detection changes view it's better to keep immutability and change object/arrays references than mutating existing objects, i.e: 
```
this. trackedTransactionHashes = this.trackedTransactionHashes.concat([transaction.transactionHash]);
```